packages/frontend/src/features/post/hooks/useBackgroundReputationClaim/useBackgroundReputationClaim.ts

```diff
@@ -0,0 +1,196 @@
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query'
+import { useUserState } from '@/features/core'
+import { MutationKeys, QueryKeys } from '@/constants/queryKeys'
+import {
+    fetchReportsWaitingForTransaction,
+    relayClaimReputation,
+} from '@/utils/api'
+import { RepUserType, ReputationType } from '@/types/Report'
+import { useEffect, useCallback } from 'react'
+import {
+    flattenProof,
+    genProofAndVerify,
+    genReportNonNullifierCircuitInput,
+    genReportNullifierCircuitInput,
+} from '@unirep-app/contracts/test/utils'
+import { genNullifier } from '@unirep-app/circuits/test/utils'
```

Comment:
You can use nullifier helpers in the frontend package instead of circuit testing utilities.
- https://github.com/social-tw/social-tw-website/blob/main/packages/frontend/src/features/reporting/utils/helpers.ts